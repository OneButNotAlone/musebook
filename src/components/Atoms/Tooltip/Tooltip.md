# Tooltip

This component displays a tooltip when the parent element is hovered or receives focus.

By default it renders as a `<div>`

## Usage Examples

```jsx

import { Button, Tooltip } from "@walmartlabs/Aether";

<Button>
  Hover me
  <Tooltip>Tooltip</Tooltip>
</Button>

```

Multiple Tooltips are possible

```jsx

import { Button, Tooltip } from "@walmartlabs/Aether";

<Button margin="50px 80px">
  Hover me
  <Tooltip placement="top"><Tooltip.Arrow />Tooltip</Tooltip>
  <Tooltip placement="right"><Tooltip.Arrow />Tooltip</Tooltip>
  <Tooltip placement="bottom"><Tooltip.Arrow />Tooltip</Tooltip>
  <Tooltip placement="left"><Tooltip.Arrow />Tooltip</Tooltip>
</Button>

```