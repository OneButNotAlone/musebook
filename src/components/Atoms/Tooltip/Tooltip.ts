import { Tooltip as reakit_tooltip } from 'reakit';
import { default as styled } from 'styled-components';

const StyledTooltip = styled(reakit_tooltip)`
    font-size: 14px;
    font-weight: 500;
    background-color: rgb(0, 0, 0);
    color: rgb(255, 255, 255);
    text-align: center;
    padding: 12px;
`;

const StyledArrow = styled(reakit_tooltip.Arrow)``;

export const Tooltip = Object.assign(StyledTooltip, { Arrow: StyledArrow });
