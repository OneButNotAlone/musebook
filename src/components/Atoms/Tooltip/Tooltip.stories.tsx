import * as React from 'react';
import { storiesOf } from '@storybook/react';

import { Tooltip } from './';

import * as ToolTipMarkdown from './Tooltip.md';

storiesOf('Tooltip', module).add(
    'Default',
    () => (
        <button style={{ margin: '100px 100px' }}>
            Button Text
            <Tooltip placement="top">
                <Tooltip.Arrow />
                Tooltip
            </Tooltip>
            <Tooltip placement="right">
                <Tooltip.Arrow />
                Tooltip
            </Tooltip>
            <Tooltip placement="bottom">
                <Tooltip.Arrow />
                Tooltip
            </Tooltip>
            <Tooltip placement="left">
                <Tooltip.Arrow />
                Tooltip
            </Tooltip>
        </button>
    ),
    { notes: ToolTipMarkdown }
);
