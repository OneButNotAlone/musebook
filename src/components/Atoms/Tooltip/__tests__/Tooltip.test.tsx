import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Tooltip } from '../Tooltip';

describe('Tooltip component', () => {

    test('Tooltip exists', () => {
        expect(Tooltip).toBeDefined();
    })

    
    test('Tooltip styles match snapshot', () => {
        const tree = renderer.create(
            <Tooltip />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        })

    test('Tooltip.Arrow exists', () => {
        expect(Tooltip.Arrow).toBeDefined();
    })
        
    test('Tooltip.Arrow styles match snapshot', () => {
        const tree = renderer.create(
            <Tooltip.Arrow />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })

})