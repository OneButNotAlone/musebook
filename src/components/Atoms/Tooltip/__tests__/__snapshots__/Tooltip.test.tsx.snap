// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component Tooltip styles match snapshot 1`] = `
.c3 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c2 {
  -webkit-transform: translate3d(0px,0px,0px);
  -ms-transform: translate3d(0px,0px,0px);
  transform: translate3d(0px,0px,0px);
}

.c2[aria-hidden="true"] {
  pointer-events: none;
  display: none !important;
}

.c1 {
  position: absolute;
  top: 0;
  left: 0;
  -webkit-user-select: auto;
  -moz-user-select: auto;
  -ms-user-select: auto;
  user-select: auto;
  cursor: auto;
  z-index: 999;
}

.c0 {
  font-size: 14px;
  font-weight: 500;
  background-color: rgb(0,0,0);
  color: rgb(255,255,255);
  text-align: center;
  padding: 12px;
}

<div
  aria-hidden={true}
  className="c0 c1 c2 c3"
  data-placement="top"
  hidden={true}
  onTransitionEnd={[Function]}
  role="tooltip"
/>
`;

exports[`Tooltip component Tooltip.Arrow styles match snapshot 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  position: absolute;
  font-size: 30px;
  width: 1em;
  height: 1em;
  pointer-events: none;
  background-color: transparent;
}

.c0 .stroke {
  fill: transparent;
}

.c0 .fill {
  fill: unset;
}

[data-placement^="top"] > .c0 {
  top: 100%;
  -webkit-transform: rotateZ(180deg);
  -ms-transform: rotateZ(180deg);
  transform: rotateZ(180deg);
}

[data-placement^="right"] > .c0 {
  right: 100%;
  -webkit-transform: rotateZ(-90deg);
  -ms-transform: rotateZ(-90deg);
  transform: rotateZ(-90deg);
}

[data-placement^="bottom"] > .c0 {
  bottom: 100%;
  -webkit-transform: rotateZ(360deg);
  -ms-transform: rotateZ(360deg);
  transform: rotateZ(360deg);
}

[data-placement^="left"] > .c0 {
  left: 100%;
  -webkit-transform: rotateZ(90deg);
  -ms-transform: rotateZ(90deg);
  transform: rotateZ(90deg);
}

<div
  className="c0 c1"
  style={
    Object {
      "strokeColor": "transparent",
    }
  }
>
  <svg
    viewBox="0 0 30 30"
  >
    <path
      className="stroke"
      d="M23.7,27.1L17,19.9C16.5,19.3,15.8,19,15,19s-1.6,0.3-2.1,0.9l-6.6,7.2C5.3,28.1,3.4,29,2,29h26 C26.7,29,24.6,28.1,23.7,27.1z"
    />
    <path
      className="fill"
      d="M23,27.8c1.1,1.2,3.4,2.2,5,2.2h2H0h2c1.7,0,3.9-1,5-2.2l6.6-7.2c0.7-0.8,2-0.8,2.7,0L23,27.8L23,27.8z"
    />
  </svg>
</div>
`;
